container.go

```diff
@@ -222,7 +222,7 @@ func (c *Container) fetchMounts() ([]Mount, error) {
 }
 
 // fetchContainer fetches a container config from a pod ID and returns a Container.
-func fetchContainer(pod *Pod, containerID string) (*Container, error) {
+func fetchContainer(pod *Pod, containerID string) (VCContainer, error) {
```

Comment:
I don't think this function needs to return `VCContainer` type since it is only used from the virtcontainers implementation. Neither the mock implementation, nor any external consumer of the lib can access this function as it is private. Leaving this function unchanged simplifies the changes above (in `api.go`).